controllers/ibu_controller.go

```diff
@@ -474,6 +426,15 @@ func (r *ImageBasedUpgradeReconciler) SetupWithManager(mgr ctrl.Manager) error {
 					return true
 				}
 
+				// trigger reconcile upon adding or updating TriggerReconcileAnnotation
+				oldValue, oldExist := e.ObjectOld.GetAnnotations()[utils.TriggerReconcileAnnotation]
+				newValue, newExist := e.ObjectNew.GetAnnotations()[utils.TriggerReconcileAnnotation]
+				if oldExist && !newExist {
+					return false
+				} else if !oldExist && newExist || oldValue != newValue {
+					return true
+				}
```

Comment:
i wonder if this could be simplified a bit? 

I thinking return early with false if reflect.DeepEqual(e.ObjectNew.GetAnnotations(), e.ObjectOld.GetAnnotations())

otherwise return oldExist != newExist?

Reply:
`or updating TriggerReconcileAnnotation` as in removing yeah?

Reply:
yeah, right, so adding/removing `ManualCleanupAnnotation` can trigger reconcile, but we want a dedicated annotation for this particular purpose when really bad things happen. And we want to limit the reconciliation to these two specific annotations only.

Getting annotations and comparing should be super pretty quick so I don't see the necessity to add additional check to `return early with false if reflect.DeepEqual(e.ObjectNew.GetAnnotations(), e.ObjectOld.GetAnnotations())`.  

I update the check for TriggerReconcileAnnotation to one line. It should be clearer.